### ISSUE TITLE
Embroider (v0.35) + Fastboot SSR exception - "document is not defined"

### ISSUE THREAD

**Issue by lukecoy** [embroider-build/embroider#664]
Repo with issue fully reproduced [here](https://github.com/lukecoy/ember-embroider-fastboot-issue#issue)

In certain scenarios, Webpack seems to call `document.createElement` in Fastboot, which causes reference errors when serving apps locally. I didn't see this on Embroider `0.28.0`, but did when upgrading to `0.35.0`. This seems to be only reproducible once you *have* integration tests added - If you delete the integration tests in my example repo, this error oddly goes away! Exception below:

```
ReferenceError: document is not defined
    at Function.requireEnsure [as e] (/var/folders/jl/71dybhnj1zvcmkhj50vmjzbw0000gn/T/broccoli-8738901D5cd9H2gma/out-253-packager_runner_embroider_webpack/assets/chunk.5b7e63d4f16ef67e9215.js:116:27)
    at Object.load (webpack:///./assets/ember-embroider-fastboot-issue.js?:104:60)
    at PrivateRouter.name [as getRoute] (webpack:///./node_modules/@embroider/router/index.js?:87:23)
    at UnresolvedRouteInfoByParam.fetchRoute (/var/folders/jl/71dybhnj1zvcmkhj50vmjzbw0000gn/T/broccoli-8738901D5cd9H2gma/out-253-packager_runner_embroider_webpack/assets/router_js.js:830:1)
...
```

**Comment by rahulk94** [comment]
I _think_ this should be fixed in 1.1.0 by https://github.com/embroider-build/embroider/pull/1109

**Comment by ef4** [comment]
Yes, and also the bug fixed in #1140 could also results in errors like this one sometimes. I expect that one or both of those fixed this, but please reopen if it happens on embroider 1.5
